lms/services/moodle.py

```diff
@@ -137,14 +137,15 @@ def list_files(self, course_id: int):
     def file_exists(self, file_id) -> bool:
         """Check if the file exists in the course."""
         # Moodle file IDs are URLs, but they need the token to be accessible
-        response = self._http.request("HEAD", f"{file_id}&token={self.token}")
+        response = self._http.request(
+            "GET", f"{file_id}&token={self.token}", headers={"Range": "bytes=0-5"}
+        )
         # Moodle API doesn't use status codes, we can't rely on that.
-        # We don't want to download the full file so we'll do a HEAD request and assume:
-        #   - JSON response, it's an error response
-        #   - Anything else, it's  the file we are after
-
-        LOG.info("Headers from Moodle file check %s", response.headers)
-        return not response.headers["content-type"].startswith("application/json")
+        # We don't want to download the full file so we'll do a GET request for the first bytes
+        # and check if the files is indeed a PDF.
+        # Most error response end up being JSON but not all of them are, some are HTML.
+        LOG.info("Content from Moodle file check %s", response.text)
+        return "%PDF" in response.text
```

Comment:
Oh, right. UTF-8 is not actually the default assumed encoding because these are old standards. I'm not clear if `response.text` can fail, but nevertheless treating the response as binary would be the correct thing to do.